vespajlib/src/main/java/com/yahoo/geo/OneDegreeParser.java

```diff
@@ -0,0 +1,282 @@
+
+package com.yahoo.geo;
+
+/**
+ * utility for parsing one geographical coordinate
+ *
+ * @author arnej27959
+ **/
+class OneDegreeParser {
+    /**
+     * the parsed latitude (degrees north if positive)
+     **/
+    public double latitude = 0;
+    public boolean foundLatitude = false;
+
+    /**
+     * the parsed longitude (degrees east if positive)
+     **/
+    public double longitude = 0;
+    public boolean foundLongitude = false;
+
+    public static boolean isDigit(char ch) {
+        return (ch >= '0' && ch <= '9');
+    }
+    public static boolean isCompassDirection(char ch) {
+        return (ch == 'N' || ch == 'S' || ch == 'E' || ch == 'W');
+    }
+
+    private String parseString = null;
+    private int len = 0;
+    private int pos = 0;
+
+    public String toString() {
+        if (foundLatitude) {
+            return parseString + " -> latitude(" + latitude + ")";
+        } else {
+            return parseString + " -> longitude(" + longitude + ")";
+        }
+    }
+
+    private char getNextChar() throws IllegalArgumentException {
+        if (pos == len) {
+            pos++;
+            return 0;
+        } else if (pos > len) {
+            throw new IllegalArgumentException("position after end of string");
+        } else {
+            return parseString.charAt(pos++);
+        }
+    }
+
+    /**
+     * Parse the given string.
+     *
+     * The string must contain either a latitude or a longitude.
+     * A latitude should contain "N" or "S" and a number signifying
+     * degrees north or south, or a signed number.
+     * A longitude should contain "E" or "W" and a number
+     * signifying degrees east or west, or a signed number.
+     * <br>
+     * Fractional degrees are recommended as the main input format,
+     * but degrees plus fractional minutes may be used for testing.
+     * You can use the degree sign (U+00B0 as seen in unicode at
+     * http://www.unicode.org/charts/PDF/U0080.pdf) to separate
+     * degrees from minutes, put the direction (NSEW) between as a
+     * separator, or use a small letter 'o' as a replacement for the
+     * degrees sign.
+     * <br>
+     * Some valid input formats: <br>
+     * "37.416383" and "-122.024683" → Sunnyvale <br>
+     * "N37.416383" and "W122.024683" → Sunnyvale <br>
+     * "37N24.983" and "122W01.481" → same <br>
+     * "N37\u00B024.983" and "W122\u00B001.481" → same <br>
+     * "63.418417" and "10.433033" → Trondheim <br>
+     * "N63.418417" and "E10.433033" → same <br>
+     * "N63o25.105" and "E10o25.982" → same <br>
+     * "E10o25.982" and "N63o25.105" → same <br>
+     * "N63.418417" and "E10.433033" → same <br>
+     * "63N25.105" and "10E25.982" → same <br>
+     * @param assume_n_s Latitude assumed, otherwise longitude
+     * @param lat_or_lon Latitude or longitude
+     *
+     **/
+    public OneDegreeParser(boolean assume_n_s, String lat_or_lon) throws IllegalArgumentException {
```

Comment:
Prefer camelCase for Java code. Applies for entire file.

Reply:
fixed.